### PR TITLE
Added ModuleModel & Project Management window implementation

### PR DESCRIPTION
**Pull request by fredska**
Integration of Project Management window with the new ModuleModel data object;
Begun review of how to handle Module data & interaction with NPC component;
Project Management Save & Load functions working;

### CODE REVIEW

File: FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs [hunk 1]
```diff
@@ -19,13 +10,80 @@ namespace FantasyModuleParser.Main
...
+            DataContext = projectManagementViewModel;
         }
         private void ESExit_Click(object sender, RoutedEventArgs e)
         {
             this.Close();
         }
+
+        private void btn_OpenModuleFilePath(object sender, RoutedEventArgs e)
+        {
+            FolderBrowserDialog fbd = new FolderBrowserDialog();
```
Comment (lorddusk) [inline FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs:27]: fbd to folderBrowserDialog, don't abbreviate variables.

File: FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs [hunk 2]
```diff
@@ -19,13 +10,80 @@ namespace FantasyModuleParser.Main
...
+        }
+        private void SaveModuleAndClose_Click(object sender, RoutedEventArgs e)
+        {
+            SaveModule_Click(sender, e);
+            ESExit_Click(sender, e);
+        }
+
+        private void SaveToModule_Click(object sender, RoutedEventArgs e)
+        {
+            FolderBrowserDialog fbd = new FolderBrowserDialog();
```
Comment (lorddusk) [inline FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs:61]: same as above

File: FantasyModuleParser/NPC/Views/PreviewNPC.xaml.cs [hunk 3]
```diff
@@ -22,20 +26,58 @@ namespace FantasyModuleParser.NPC.Views
...
-            //npcController = new NPCController();
-            //DataContext = npcController.GetNPCModel();
             viewModel = new PreviewNPCViewModel();
+            viewModel.NPCModel.PropertyChanged += RefreshDataContext;
+
+            setupPropertyChangedEventForLanguages(viewModel.NPCModel.StandardLanguages);
+            setupPropertyChangedEventForLanguages(viewModel.NPCModel.ExoticLanguages);
+            setupPropertyChangedEventForLanguages(viewModel.NPCModel.MonstrousLanguages);
+            // Custom method for UserLanguages, as the user can add / remove languages at will
+            viewModel.NPCModel.UserLanguages.CollectionChanged += (sender, e) =>
```
Comment (lorddusk) [inline FantasyModuleParser/NPC/Views/PreviewNPC.xaml.cs:41]: I'd prefer renaming e to event, but it's fine.

File: FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs (hunk 1 above)
Comment (EngineerSuite2) [reply FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs:27]: Resolved in PR 46

File: FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs (hunk 2 above)
Comment (EngineerSuite2) [reply FantasyModuleParser/Main/Views/ProjectManagement.xaml.cs:61]: Resolved in PR 46

**Pull request merged**